lib/App/Critique/Command/collect.pm

```diff
@@ -42,14 +42,7 @@ sub execute {
 
     my $filter;
     if ( my $f = $opt->filter ) {
-        if ( ref $f eq 'CODE' ) {
-            $filter = $f;
-        }
-        else {
-            $filter = $opt->invert
-                ? sub { $_[0]->stringify !~ /$f/ }
-                : sub { $_[0]->stringify =~ /$f/ };
-        }
+        $filter = file_filter($opt,'file_filter_regex');
     }
```

Comment:
the word 'default' is probably a little misleading, its actually more of a 'fallback', so if the filter is not a coderef, then it falls back to 'file_filter_regex', this is why named params would be make more sense. As this could be `$filter = file_filter(%$opt, fallback_filter => 'file_filter_regex');`